modules/programs/vdirsyncer-accounts.nix

```diff
@@ -0,0 +1,217 @@
+      type = types.nullOr (types.listOf types.str);
+      default = null;
+      description = ''
+        Kinds of items to show. The default is to
+        show everything. This depends on particular
+        features of the server, the results are not
+        validated.
+      '';
+    };
+
+    verify = mkOption {
+      type = types.nullOr types.bool;
+      default = null;
+      description = "Verify SSL certificate.";
+    };
+
+    verifyFingerprint = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        Optional. SHA1 or MD5 fingerprint of the expected server certificate.</para>
+
+        <para>See
+        <link xlink:href="https://vdirsyncer.pimutils.org/en/stable/ssl-tutorial.html#ssl-tutorial"/>
+        for more information.
+      '';
+    };
+
+    auth = mkOption {
+      type = types.nullOr (types.enum ["basic" "digest" "guess"]);
+      default = null;
+      description = ''
+        Authentication settings. The default is <literal>"basic"</literal>.
+      '';
+    };
+
+    authCert = mkOption {
+      type = types.nullOr (types.either types.str (types.listOf types.str));
+      default = null;
+      description = ''
+      Either a path to a certificate with a client certificate
+      and the key or a list of paths to the files with them.
+     '';
+    };
+
+    userAgent = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        The user agent to report to the server.
+        Defaults to <literal>"vdirsyncer"</literal>.
+      '';
+    };
+
+    postHook = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        Command to call for each item creation and modification.
+        The command will be called with the path of the new/updated
+        file.
+      '';
+    };
+
+    ## Options for google storages
+
+    tokenFile = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        A file path where access tokens
+        are stored.
+      '';
+    };
+
+    clientId = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        OAuth credentials, obtained from the Google API Manager.</para>
+
+        <para> See
+        <link xlink:href="https://vdirsyncer.pimutils.org/en/stable/config.html#google"/>
+        for more information.
+      '';
+    };
+
+    clientIdCommand = mkOption {
+      type = types.nullOr (types.listOf types.str);
+      default = null;
+      example = [ "pass" "client_id" ];
+      description = ''
+        A command that prints the OAuth credentials to standard
+        output.
+
+        OAuth credentials, obtained from the Google API Manager.</para>
+
+        <para> See
+        <link xlink:href="https://vdirsyncer.pimutils.org/en/stable/config.html#google"/>
+        for more information.
+      '';
+    };
+
+    clientSecret = mkOption {
```

Comment:
Similarly, can remove so that the users are encouraged to use `clientSecretCommand` instead.